libs/application/templates/estate/src/fields/VehicleRepeater/index.tsx

```diff
@@ -0,0 +1,140 @@
+import { FC, useEffect } from 'react'
+import { useFieldArray, useFormContext } from 'react-hook-form'
+import { useLocale } from '@island.is/localization'
+import { FieldBaseProps } from '@island.is/application/types'
+import {
+  Box,
+  GridColumn,
+  GridRow,
+  Button,
+  ProfileCard,
+} from '@island.is/island-ui/core'
+import { Answers, AssetFormField } from '../../types'
+
+import { m } from '../../lib/messages'
+import { EstateAsset } from '@island.is/clients/syslumenn'
+import { AdditionalVehicle } from './AdditionalVehicle'
+import { InputController } from '@island.is/shared/form-fields'
+
+export const VehicleRepeater: FC<
+  React.PropsWithChildren<FieldBaseProps<Answers>>
+> = ({ application, field, errors }) => {
+  const error = (errors as any)?.estate?.vehicles
+  const { id } = field
+  const { formatMessage } = useLocale()
+  const { fields, append, remove, update, replace } = useFieldArray({
+    name: id,
+  })
+
+  const { clearErrors } = useFormContext()
+
+  const externalData = application.externalData.syslumennOnEntry?.data as {
+    estate: { vehicles: EstateAsset[] }
+  }
+
+  useEffect(() => {
+    if (fields.length === 0 && externalData.estate.vehicles) {
+      replace(externalData.estate.vehicles)
+    }
+  }, [])
+
+  const handleAddProperty = () =>
+    append({
+      share: 1,
+      assetNumber: undefined,
+      description: undefined,
+      marketValue: undefined,
+      initial: false,
+      enabled: true,
+    })
+  const handleRemoveProperty = (index: number) => remove(index)
+
+  return (
+    <Box marginTop={2}>
+      <GridRow>
+        {fields.reduce((acc, vehicle: AssetFormField, index) => {
```

Comment:
might want to check if fields has length before calling reduce
```suggestion
        {fields.length > 0 && fields.reduce((acc, vehicle: AssetFormField, index) => {
```

Reply:
Safety first :vulcan_salute: